validation.md

```diff
@@ -2241,6 +2244,211 @@ Occasionally, you may want to attach additional validation rules to your default
+
+  // Allow some Unicode
+  Rule::email()->withNativeValidation(allowUnicode: true);
+```
+
+<a name="email-validation-rule-matrix"></a>
+### Email Validation Rule Matrix
+
+Below is a single table showcasing different email address formats, highlighting whether each ✅ **passes** or ❌ **fails** the validation methods:
+
+| **Email Address**                    | **rfcCompliant()** | **rfcCompliant(strict: true)** | **withNativeValidation()** | **withNativeValidation(allowUnicode: true)** | **validateMxRecord()** | **preventSpoofing()** |
+|--------------------------------------|--------------------|--------------------------------|----------------------------|----------------------------------------------|------------------------|-----------------------|
+| `"has space"@example.com`            | ✅                  | ❌                              | ❌                          | ❌                                            | ✅                      | ✅                     |
+| `abc."test"@example.com`             | ✅                  | ❌                              | ✅                          | ✅                                            | ✅                      | ✅                     |
+| `"ab\\(c"@example.com`               | ✅                  | ❌                              | ✅                          | ✅                                            | ✅                      | ✅                     |
+| `name@[127.0.0.1]`                   | ✅                  | ❌                              | ✅                          | ✅                                            | ✅                      | ✅                     |
+| `test@example` <br>(no TLD)          | ✅                  | ❌                              | ❌                          | ❌                                            | ❌                      | ✅                     |
+| `test@localhost` <br>(no TLD)        | ✅                  | ❌                              | ❌                          | ❌                                            | ❌                      | ✅                     |
+| `plainaddress@example.com`           | ✅                  | ✅                              | ✅                          | ✅                                            | ✅                      | ✅                     |
+| `tést@example.com`                   | ✅                  | ✅                              | ❌                          | ✅                                            | ✅                      | ✅                     |
+| `user@üñîçødé.com`                   | ✅                  | ✅                              | ❌                          | ❌                                            | ✅                      | ✅                     |
+| `admin@examрle.com` <br>(Cyrillic р) | ✅                  | ✅                              | ✅                          | ✅                                            | ✅                      | ❌                     |
```

Comment:
I think it would be helpful to clarify why admin@examрle.com (with Cyrillic р) passes rfcCompliant(strict: true) but fails preventSpoofing(). Since it’s a potentially deceptive address

Reply:
Because it's a valid email address, just deceptive so unwanted. So the RFC has nothing to do with how. I am not sure how to add these kind of details without making the docs for email validation a book on itself 😄